src/ripple/protocol/impl/Serializer.cpp

```diff
@@ -470,29 +445,35 @@ bool Serializer::getVLLength (int& length, int offset) const
     return true;
 }
 
-Blob Serializer::encodeVL (int length)
+int Serializer::addEncoded (int length)
 {
-    unsigned char lenBytes[4];
+    std::array<std::uint8_t, 4> result;
+    int l = encodeVL (length, result);
+    return addRaw (&result[0], l);
+}
 
+int Serializer::encodeVL (int length,
+                          std::array<std::uint8_t, 4>& result)
+{
     if (length <= 192)
     {
-        lenBytes[0] = static_cast<unsigned char> (length);
-        return Blob (&lenBytes[0], &lenBytes[1]);
+        result[0] = static_cast<unsigned char> (length);
+        return 1;
     }
     else if (length <= 12480)
```

Comment:
What are these magic numbers?  I'll look around to see...

Quibble: you don't need an else if the previous clause ends with a return.